### PR TITLE
Put fast-auth-signer in iframe for authentication process.

### PR DESCRIPTION
**Pull request by benmalcom**
This PR makes some little changes to `DesktopNavigation`, `useSignInRedirect` and `vmInitializer` files to initiate our POC, which is an  iframe/modal based authentication instead of the page-based authentication method we currently have.

cc: @esaminu

### CODE REVIEW

File: src/hooks/useSignInRedirect.ts [hunk 1]
```diff
@@ -29,9 +31,17 @@ export function useSignInRedirect() {
...
-      router.push(createAccount ? '/signup' : '/signin');
+      if (!vmNear) return;
+      vmNear.selector
+        .then((selector: any) => selector.wallet('fast-auth-wallet'))
+        .then((fastAuthWallet: any) =>
+          fastAuthWallet.signIn({
+            contractId: vmNear.config.contractName,
+            isRecovery: !createAccount,
+          }),
+        );
```
Comment (calebjacob) [inline src/hooks/useSignInRedirect.ts:42]: These changes completely break our `Sign In` and `Create Account` buttons in the build preview. We can't remove this logic:

```
router.push(createAccount ? '/signup' : '/signin');
```

Without directing the user to these pages, they won't be able to choose if they want to sign in with FastAuth or with their wallet. We can't assume users will only sign in with FastAuth.

@charleslavon do you have any context on these changes?

**Review by calebjacob — CHANGES_REQUESTED** [review]

**Comment by charleslavon** [comment]
Hey @esaminu what intended change will this PR have on UX?

**Comment by esaminu** [comment]
@charleslavon @calebjacob These changes will still allow users to access other wallets via the "Connect Wallet" button in the iframe which will open wallet selector by redirecting with a query param `connectWallet` set to true which will show the wallet selector in this implementation. [Here's the figma for this stream of work](https://www.figma.com/file/oK9aMWVmjoXHRUcBUFmFvT/Onboarding?type=design&node-id=4178-165620&mode=design&t=jKr6zGEkshBAZ3ZX-0). It also require a version bump once we merge https://github.com/near/near-fastauth-wallet/pull/14 so we shouldn't merge until that is published and also after https://github.com/near/fast-auth-signer/pull/155 is merged

cc @nall-near

**Review by esaminu — APPROVED** [review]
LGTM

File: src/components/vm/VmInitializer.tsx [hunk 2]
```diff
@@ -142,6 +145,14 @@ export default function VmInitializer() {
     });
   }, [idOS, near]);
 
+  const handleWalletSelectorMessage = (e: MessageEvent<{ showWalletSelector: boolean }>) => {
```
Comment (charleslavon) [inline src/components/vm/VmInitializer.tsx:148]: is this necessary? It doesn't appear to be used

File: src/components/vm/VmInitializer.tsx [hunk 3]
```diff
@@ -60,6 +61,7 @@ export default function VmInitializer() {
   const setVmStore = useVmStore((store) => store.set);
   const { requestAuthentication, saveCurrentUrl } = useSignInRedirect();
   const idOS = useIdOS();
+  const router = useRouter();
```
Comment (charleslavon) [inline src/components/vm/VmInitializer.tsx:64]: Kindly remove this and its import if it's not necessary.

Comment (benmalcom) [reply src/components/vm/VmInitializer.tsx]: fixed

**Review by charleslavon — APPROVED** [review]

**Pull request merged**